### DISCUSSION TITLE
Filtering Data on Linked Screen When Using External Database (MS SQL)

### DISCUSSION THREAD

**Discussion by chadgriff** [Budibase/budibase#2946] [Help] [answered]
I am having trouble getting a filter to work on a linked screen. I can get it to work when I load sample data into the internal Budibase database (CouchDB) but cannot get it work as expected when the data resides in my MS SQL database. My use case is that I need to show a list of commission records and be able to update the "paid" status (similar to paying invoice).  Since I cannot get the initial screen to refresh after a button click, I was attempting to navigate to another screen, update the status, and then navigate back (in order to refresh the table).  However, I cannot seem to get the filter working on the downstream screen when pulling the data via a custom SQL query. Any guidance would be appreciated.
![Link1](https://user-images.githubusercontent.com/15015476/136662056-d0e16fc4-9345-48de-9bde-9055b75609d9.png)
![link2](https://user-images.githubusercontent.com/15015476/136662067-d6bcb9eb-fa5f-4513-8e6c-e95567ee5d58.png)

**Comment by dncpax** [comment, marked as answer]
I have the same problem - filtering does not work for my postgresql tables (schema not public). 
My work around is to create a query for the screen with a parameter. Than I use this query in the new screen and use the :id screen parameter as the parameter for the query.
This works fine for me...

Setting the query with parameter:
![image](https://user-images.githubusercontent.com/2836636/136694040-625f954b-ea88-4e96-8d44-6bdbab247fd8.png)

Setting the new screen datasource with the screen parameter:
![image](https://user-images.githubusercontent.com/2836636/136694111-56a42b3d-f35c-452e-b34b-366b40bcf9d2.png)

**Reply by chadgriff** [reply]
@dncpax Thanks for the idea. This will suit my use case perfectly.
